frontend/components/puzzle2.jsx

```diff
@@ -1,8 +1,35 @@
 import React from 'react';
+import { Route, Redirect } from 'react-router-dom';
 
 // Answer: See o double you
 
 class Puzzle2 extends React.Component {
+  constructor() {
+    super();
+    
+   this.state = {
+      puzzlesolved: 'no',
+    };
+     
+    }
+
+  submitanswer() {
+    var input = document.getElementById('puzzleAnswer').value
+    console.log(input)
+
+    sendresponse = this.props.send('submit', {puzzle2:input})
+ // socketIO listen for the response and run a function to log response
+    var serverresponse = socket.on('submit_response', function(response) {
```

Comment:
lets put this listener in the constructor so that we don't run into timing issues where the response comes back faster than we can register this callback